Services/Administration/OJS.Services.Administration.Business/SubmissionTypes/SubmissionTypesBusinessService.cs

```diff
@@ -1,24 +1,197 @@
 
+using FluentExtensions.Extensions;
 using Microsoft.EntityFrameworkCore;
+using OJS.Data.Models;
+using OJS.Data.Models.Contests;
 using OJS.Data.Models.Submissions;
 using OJS.Services.Administration.Data;
 using OJS.Services.Administration.Models.SubmissionTypes;
+using OJS.Services.Infrastructure;
 using OJS.Services.Infrastructure.Exceptions;
 using OJS.Services.Infrastructure.Extensions;
+using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Text;
 using System.Threading.Tasks;
+using System.Transactions;
+using static OJS.Common.GlobalConstants.Roles;
 
 public class SubmissionTypesBusinessService : AdministrationOperationService<SubmissionType, int, SubmissionTypeAdministrationModel>, ISubmissionTypesBusinessService
 {
     private readonly ISubmissionTypesDataService submissionTypesDataService;
+    private readonly IContestsDataService contestsDataService;
+    private readonly ISubmissionsDataService submissionsDataService;
+    private readonly IParticipantScoresDataService participantScoresData;
+    private readonly ITestRunsDataService testRunsData;
+    private readonly ISubmissionTypesInProblemsDataService submissionTypesInProblemsDataService;
+    private readonly IDatesService datesService;
 
-    public SubmissionTypesBusinessService(ISubmissionTypesDataService submissionTypesDataService)
-        => this.submissionTypesDataService = submissionTypesDataService;
+    public SubmissionTypesBusinessService(
+        ISubmissionTypesDataService submissionTypesDataService,
+        IContestsDataService contestsDataService,
+        ISubmissionsDataService submissionsDataService,
+        IParticipantScoresDataService participantScoresData,
+        ITestRunsDataService testRunsData,
+        ISubmissionTypesInProblemsDataService submissionTypesInProblemsDataService,
+        IDatesService datesService)
+    {
+        this.submissionTypesDataService = submissionTypesDataService;
+        this.contestsDataService = contestsDataService;
+        this.submissionsDataService = submissionsDataService;
+        this.participantScoresData = participantScoresData;
+        this.testRunsData = testRunsData;
+        this.submissionTypesInProblemsDataService = submissionTypesInProblemsDataService;
+        this.datesService = datesService;
+    }
 
     public async Task<List<SubmissionTypesInProblemView>> GetForProblem() =>
         await this.submissionTypesDataService.GetAll().MapCollection<SubmissionTypesInProblemView>().ToListAsync();
 
+    public async Task<string> ReplaceSubmissionType(ReplaceSubmissionTypeServiceModel model)
+    {
+        var stringBuilder = new StringBuilder();
+
+        if (model.SubmissionTypeToReplaceWith.HasValue && model.SubmissionTypeToReplace == model.SubmissionTypeToReplaceWith.Value)
+        {
+            throw new BusinessServiceException("Cannot replace submission type with identical submission type");
+        }
+
```

Comment:
Extract into Validator along with the submissionsByRegularUsersInTheLastMonth logic